### DISCUSSION TITLE
How to compute transfer timings from a curl_easy handle

### DISCUSSION THREAD

**Discussion by alexsn** [curl/curl#14184] [General]
Hey guys,

I'd like to compute the following stats from a curl transfer:
`firstByteFlushed` - Time from the start until the first protocol byte is sent (excluding connect times).
`lastByteFlushed` - Time from the start until the last protocol byte is sent, i.e. finished sending the request (excluding connect times).
`timeToFirstByte` - Time from the start until the first response byte is received  (excluding connect times).
`timeToLastByte` - Time from the start until the last response byte is received  (excluding connect times).

I'm able to compute `firstByteFlushed` from `CURLINFO_PRETRANSFER_TIME_T` and `timeToFirstByte` from `CURLINFO_STARTTRANSFER_TIME_T`. How can I compute the other two, one idea is to use `CURLOPT_DEBUGFUNCTION`, another is to use `CURLOPT_XFERINFOFUNCTION`. Will those work for what I want, are there better ways to achieve the above?

**Comment by alexsn** [comment]
I mainly would like to know when [this](https://github.com/curl/curl/blob/master/lib/request.c#L257) happend, @bagder is it possible to add `CURLINFO_UPLOADDONE_TIME_T`? If so I'd be happy to implement it.

**Reply by bagder** [reply]
I can't think of any good way to figure that out in any exact way right now. I believe adding and providing such a timestamp CURLINFO* variable can be valuable for users, yes.

**Reply by alexsn** [reply]
See https://github.com/curl/curl/pull/14189
